src/prefect/tasks/azure/datafactory.py

```diff
@@ -0,0 +1,356 @@
+import time
+from datetime import datetime, timedelta
+from typing import Any, Dict, List, Optional
+
+import azure.identity
+import azure.mgmt.datafactory
+
+from prefect import Task
+from prefect.client import Secret
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+def _get_datafactory_client(azure_credentials_secret: str):
+    """
+    Helper function to create datafactory client.
+    """
+    azure_credentials = Secret(azure_credentials_secret).get()
```

Comment:
I'd recommend passing in the full credentials and move the `Secret.get()` call into the tasks to make this function more flexible.

Reply:
Can you elaborate on what you mean by "full_credentials"?

Reply:
I think in this case it'd be a dictionary containing `client_id`, `client_secret`, and `tenant_id` keys